### PR TITLE
Expert metalurgy

### PR DESCRIPTION
**Pull request by MuteTiefling**
## Crushing
Crushing happens in tiers, with later materials only being able to be crushed by better tech. 
This includes Gems, Ingots, Gears, and Plates to Dusts. 

Tier 1 Crushing: Hexerei Pestle and Mortar
Materials: 
- apatite
- bronze
- charcoal
- coal
- coal_coke
- copper
- electrum
- gold
- lapis
- niter
- silver
- sulfur
- tin

Tier 2 Crushing: Create Millstone and Ars Nouveau Crush Glyph
Materials: 
- amethyst
- certus_quartz
- constantan
- dimensional
- fluix
- fluorite
- lead
- nickel
- quartz

Tier 3 Crushing: Create Crushing Wheel
Materials: 
- aluminum
- diamond
- emerald
- ender_pearl
- falsite
- horizonite
- invar
- iron
- osmium
- ventium

Tier 4 Crushing: Immersive Engineering Crusher and Thermal Pulverizer (Plus Thermal Earth Charges for Gems only)
Materials: 
- enderium
- iesnium
- lumium
- netherite
- signalum
- steel
- uranium

## Ores and Secondary Outputs

|   Primary  | Secondary |
|:----------:|:---------:|
|   Copper   |   Nickel  |
|   Silver   |    Lead   |
|    Gold    |   Copper  |
|     Tin    |   Quartz  |
| Horizonite |   Copper  |
|   Ventium  |    Nickel   |
|   Falsite  |    Lead   |
|    Iron    |    Tin    |

### CODE REVIEW

**Comment by MuteTiefling** [comment]
# Ore Processing - Simple Metals
Simple metals include Copper, Silver, Gold, Tin, Horizonite, Ventium, and Falsite. 
These ores all follow the same mechanics for processing. 

## Tier 1
Basic Crushing occurs in the Pestle and Mortar and produces little output. Higher tiers of crushing give better results, however.
For all of these ores, there's a chance to obtain the raw ore item back again, allowing it to be passed through for another crush and more output. The average number of times this may occur depends on crushing tier. 
image
image
image

From here, crushed ore may be smelted in a blast furnace and other powered furnaces.
image
The resulting Clump is then crushed again
image
And finally smelted to an Ingot
image


## Tier 2
Tier 2 also begins with Crushing as above. However the resulting Crushed Ore may be bathed in acid to obtain some extra results. 
image
image
The resulting Dirty Dust is then smelted as above to an Ingot

**Comment by MuteTiefling** [comment]
# Ore Processing - Iron
Changes to original Iron process.

Crush the Ore. First option for this is Create as before. 
image

Next step requires superheating
image

Blast to Crude Iron
image

Infuse to Iron
image

**Comment by MuteTiefling** [comment]
## Osmium Processing
Osmium is a byproduct only, and will not spawn in world (not yet implemented)

It can be obtained from Rich Slag
image

Sequence Assembly allows similar processing, but is chance based. Should result in roughly the same output, though it also gives regular slag as the 'filler' output. 
image

Osmium Clumps are processed down with Water Essence
image

Finally, the Dirty Dust may be blasted down.
image


Rich Slag is obtained from Iron processing as shown above, as well as from Nickel processing.

**Comment by MuteTiefling** [comment]
# Nickel Processing
Nickel comes up from a couple sources as a byproduct. 

Crushed Nickel Ore can come up from Copper Ore
image

This may be Blasted to Clumps, producing some Rich Slag as well. 
image

This may be Infused, to produce some extra: 
image

And Finally blasted again for more Rich Slag and an Ingot
image

Alternatively, Nickel may be obtained by washing Copper and Ventium crushed ores with Redstone Acid, either in a Bottling Machine or with Create. 
image

**Comment by MuteTiefling** [comment]
@Kanzaji Moving EE Deposits to Config Swapper folders. I need to override some world gen for Expert as some of these ores aren't meant to spawn.

**Comment by NielsPilgaard** [comment]
> Osmium is a byproduct only, and will not spawn in world (not yet implemented)

Are we doing different oregen between normal/expert modes?

File: kubejs/server_scripts/expert/recipes/enigmatica/dust_crushing_tiers.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        return;
+    }
+
+    const id_prefix = 'enigmatica:expert/enigmatica/';
+    // Tiers range from 1-4, with 4 being the highest
+    const recipes = [
+        {
+            input_types: ['gems'],
+            material: 'ender_pearl',
+            output: 'thermal:ender_pearl_dust',
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/expert/recipes/enigmatica/dust_crushing_tiers.js:12]: Is this the ender pearl dust variant we want to use?

File: kubejs/server_scripts/normal/recipes/create/crushing.js
Comment (NielsPilgaard) [inline kubejs/server_scripts/normal/recipes/create/crushing.js:1]: Soft delete please :D

**Review by NielsPilgaard — CHANGES_REQUESTED** [review]

File: kubejs/server_scripts/expert/recipes/enigmatica/dust_crushing_tiers.js (hunk 1 above)
Comment (MuteTiefling) [reply kubejs/server_scripts/expert/recipes/enigmatica/dust_crushing_tiers.js:12]: Fair point, I've added it to EE now.

**Comment by MuteTiefling** [comment]
Quick change to Soul Steel, replaces one Iesnium with one Osmium dust
image

Recipe for Horizonite. This is probably the one metal from Blue Skies that'll get used a lot (since it's fuel), so I figure a way to make it from the more regularly available metals will be good. 
image

Also copies the Mixing Cauldron recipe for Fiery Ingots over to the Dissolution Chamber, minus the herbs. 
image

**Review by NielsPilgaard — APPROVED** [review]

**Pull request merged**